apps/bfd-pipeline/bfd-pipeline-rda-grpc/src/main/java/gov/cms/bfd/pipeline/rda/grpc/sink/direct/AbstractClaimRdaSink.java

```diff
@@ -200,10 +201,19 @@ private void updateLastSequenceNumberImpl(long lastSequenceNumber) {
   }
 
   private List<RdaChange<TClaim>> transformMessages(
-      String apiVersion, Collection<TMessage> messages) {
-    return messages.stream()
-        .map(message -> transformMessage(apiVersion, message))
-        .collect(Collectors.toList());
+      String apiVersion, Collection<TMessage> messages) throws ProcessingException {
+    var claims = new ArrayList<RdaChange<TClaim>>();
+    for (TMessage message : messages) {
+      try {
+        var change = transformMessage(apiVersion, message);
+        metrics.transformSuccesses.mark();
+        claims.add(change);
+      } catch (DataTransformer.TransformationException error) {
+        metrics.transformFailures.mark();
+        throw new ProcessingException(error, 0);
```

Comment:
Do we need a counter here to indicate how far we are through the messages list before the exception was thown?